scripts/save_y2logs

```diff
@@ -7,110 +7,140 @@
 
 usage()
 {
-    echo "Usage: $0 [<tgz-file-name>|<bzip2-file-name>]" >&2
+    echo "Usage: ${0##*/} [ARCHIVE_NAME]" >&2
     echo "" >&2
-    echo "Copies the YaST2 logs to a compressed tar archive." >&2
-    echo "If file name is missing, it is generated by the script." >&2
+    echo "Copies the YaST2 logs to a (compressed) tar archive." >&2
+    echo "If file name is missing it is generated by the script." >&2
     exit 1
 }
 
 if test -z "$1"; then
-  TARGET=`mktemp --suffix .tar.gz /tmp/y2log-XXXXXX`
-  COMPRESSION=--gzip
+  # choose some good default
+  if [ -x /usr/bin/xz ] ; then
+    SUFFIX=.tar.xz
+    COMPRESSION=xz
+  elif [ -x /usr/bin/gzip ] ; then
+    SUFFIX=.tar.gz
+    COMPRESSION=gzip
+  elif [ -x /usr/bin/bzip2 ] ; then
+    SUFFIX=.tar.bz2
+    COMPRESSION=bzip2
+  else
+    SUFFIX=.tar
+    COMPRESSION=
+  fi
+  TARGET=`mktemp --suffix $SUFFIX /tmp/y2log-XXXXXX`
 else
   case "$1" in
     -*)
 	usage
 	;;
 
+    *.tar)
+	TARGET="$1"
+	COMPRESSION=
+	;;
+
     *.tgz|*.tar.gz)
 	TARGET="$1"
-	COMPRESSION=--gzip
+	COMPRESSION=gzip
 	;;
 
-    *.tar.bz2)
+    *.tar.xz)
 	TARGET="$1"
-	COMPRESSION=--bzip2
+	COMPRESSION=xz
+	;;
 
-	if [ ! -x /usr/bin/bzip2 ]; then
-	    echo "FATAL: /usr/bin/bzip2 not available" >&2
-	    # This might easily happen in the inst-sys
-            exit 3
-	fi
+    *.tar.bz2)
+	TARGET="$1"
+	COMPRESSION=bzip2
 	;;
 
     *)
-	echo "FATAL: Uncompressed archives not supported" >&2
-	echo "Use one of: .tgz .tar.gz .tar.bz2"          >&2
+	echo "FATAL: argument is not a (compressed) tar file." >&2
+	echo "Use one of these suffixes: .tar .tgz .tar.gz .tar.xz .tar.bz2" >&2
 	exit 4
   esac
 fi
 
+# check for compression program
+if [ "$COMPRESSION" -a ! -x "/usr/bin/$COMPRESSION" ]; then
+  echo "FATAL: /usr/bin/$COMPRESSION not available" >&2
+  exit 3
+fi
+
 TMPDIR=`mktemp -d` || ( echo "FATAL: Failed to create a temporary directory" >&2; exit 5 );
-LIST=YaST2
 
-if [ -f /var/log/evms-engine.log ]; then
-    LIST="$LIST $( cd /var/log/; ls evms-engine.*)"
-fi
 
-if [ -x /bin/dmesg ]; then
-    dmesg > $TMPDIR/dmesg
-    LIST="$LIST $TMPDIR/dmesg"
-fi
+# 1. gather some information in $TMPDIR
 
-if [ -f /var/log/zypp/history ]; then
-    LIST="$LIST $( cd /var/log/; ls zypp/history*)"
+# last kernel messages, if any
+if [ -x /bin/dmesg ]; then
```

Comment:
point is that it is replacement for dmesg utility itself. At least on my pc, both return same output. Now it do not make much sense to have both, as both should return same lines.